pkg/security/tests/dns_resolver_test.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux && functionaltests
+
+// Package tests holds tests related files
+package tests
+
+// Package tests holds tests related files
+import (
+	"github.com/DataDog/datadog-agent/pkg/config/env"
+	sprobe "github.com/DataDog/datadog-agent/pkg/security/probe"
+	"github.com/stretchr/testify/assert"
+	"net"
+	"net/netip"
+	"slices"
+	"testing"
+	"time"
+)
+
+func TestDNSResolver(t *testing.T) {
```

Comment:
For a v2 - Can we add tests on `connect` to make sure the DNS context is properly added ?